src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/DoNotRaiseExceptionsInUnexpectedLocations.cs

```diff
@@ -102,7 +102,7 @@ public override void Initialize(AnalysisContext analysisContext)
                         if (thrownExceptionType != null && thrownExceptionType.DerivesFrom(exceptionType))
                         {
                             // If no exceptions are allowed or if the thrown exceptions is not an allowed one..
-                            if (methodCategory.AllowedExceptions.IsEmpty || !methodCategory.AllowedExceptions.Contains(thrownExceptionType))
+                            if (methodCategory.AllowedExceptions.IsEmpty || !methodCategory.AllowedExceptions.Any(n => thrownExceptionType.DerivesFrom(n)))
```

Comment:
For proper Fxcop compat, we'll need to use IsAssignableTo rather than DerivesFrom. IsAssignableTo is currently contained solely in https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/DefineAccessorsForAttributeArguments.cs#L95, could you please pull out the protected implementation into AnalyzerUtilities (an extension on ITypeSymbol should work). FxCop version: http://index/?query=DoNotRaise&rightProject=DesignRules&file=DoNotRaiseExceptionsInUnexpectedLocations.cs&line=116.

Reply:
Done. ITypeSymbolExtensions now uses CSharp and VB namespaces, however, in my opinion that's better than repeating the code.